### PR TITLE
Implement ml.estimate_model_memory.json

### PR DESCRIPTION
**Pull request by codebrain**
Implements: https://www.elastic.co/guide/en/elasticsearch/reference/7.x/ml-estimate-model-memory.html

### CODE REVIEW

File: src/Nest/XPack/MachineLearning/EstimateModelMemory/EstimateModelMemoryRequest.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	[JsonFormatter(typeof(VerbatimDictionaryKeysFormatter<MaxBucketCardinality, IMaxBucketCardinality, Field, long>))]
+	public interface IMaxBucketCardinality : IIsADictionary<Field, long> { }
+
+	public class MaxBucketCardinality : IsADictionaryBase<Field, long>, IMaxBucketCardinality
+	{
+		public MaxBucketCardinality() { }
+
+		public MaxBucketCardinality(IDictionary<Field, long> container) : base(container) { }
+
+		public MaxBucketCardinality(Dictionary<Field, long> container) : base(container) { }
```
Comment (russcam) [inline src/Nest/XPack/MachineLearning/EstimateModelMemory/EstimateModelMemoryRequest.cs:109]: Does it need both interface and concrete ctor overloads?

File: tests/Tests/XPack/MachineLearning/EstimateModelMemory/EstimateModelMemoryApiTests.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+		protected override LazyResponses ClientUsage() => Calls(
+			(client, f) => client.MachineLearning.EstimateModelMemory(f),
+			(client, f) => client.MachineLearning.EstimateModelMemoryAsync(f),
+			(client, r) => client.MachineLearning.EstimateModelMemory(r),
+			(client, r) => client.MachineLearning.EstimateModelMemoryAsync(r)
+		);
+
+		protected override EstimateModelMemoryDescriptor<Metric> NewDescriptor() => new EstimateModelMemoryDescriptor<Metric>();
+	}
```
Comment (russcam) [inline tests/Tests/XPack/MachineLearning/EstimateModelMemory/EstimateModelMemoryApiTests.cs:98]: Should this configure a job in integration setup, then make an assertion on the response in `ExpectResponse`?

File: src/Nest/XPack/MachineLearning/EstimateModelMemory/EstimateModelMemoryRequest.cs (hunk 1 above)
Comment (codebrain) [reply src/Nest/XPack/MachineLearning/EstimateModelMemory/EstimateModelMemoryRequest.cs:109]: Good Q, will remove the concrete

File: tests/Tests/XPack/MachineLearning/EstimateModelMemory/EstimateModelMemoryApiTests.cs (hunk 2 above)
Comment (codebrain) [reply tests/Tests/XPack/MachineLearning/EstimateModelMemory/EstimateModelMemoryApiTests.cs:98]: Will put an assertion on the response.

**Comment by codebrain** [comment]
Manually backported to master

**Pull request merged**